scanner/scanner.go

```diff
@@ -213,8 +189,16 @@ func (s *Scanner) walkAndParse() error {
 		if err != nil {
 			return err
 		}
+		if info.IsDir() && info.Name() == ".git" {
+			return filepath.SkipDir
+		}
```

Comment:
In all cases we only parse files (YAML for now), so in addition to skipping walking recursively inside of `.git/**`, we also want to continue, but ignore any directory. Obviously because of the regexes, we already make that nearly impossible, but in theory it's possible for someone to name a directory `some/dir/i-am-not-a-file.yaml/`

We should just

```
if info.IsDir() {
  return nil
}
```

We used to have those ^ in the parser functions, but as you remove it, we can move that check even earlier, centralized.